python/CONTRIBUTING.md

```diff
@@ -94,3 +94,31 @@ make test PYTEST_ARGS="--pdb"
 ```
 
 To see all available pytest arguments, run `make test PYTEST_ARGS="--help"`.
+
+# Code standards
+
+Below are the formalized conventions that we adhere to in the PyIceberg project. The goal of this is to have a common agreement on how to evolve the codebase, but also using it as guidelines for newcomers to the project.
+
+## API Compatibility
+
+We try to keep the Python public API compatible across versions. The Python official [PEP-8](https://peps.python.org/pep-0008/) defines Public methods as: _Public attributes should have no leading underscores_. This means not removing any methods without any notice, or removing or renaming any existing parameters. Adding new optional parameters is okay.
+
+If you want to remove a method, please add a deprecation notice by annotating the function using `@deprecated`:
+
+```python
+from pyiceberg.utils.deprecated import deprecated
+
+@deprecated(deprecated_in="0.1.0", removed_in="0.2.0")
```

Comment:
We may want to add a way to specify what to use instead, but this is great!

Reply:
Great suggestion, added it. Thanks!